server/controller/urserController.js

```diff
@@ -0,0 +1,54 @@
+const userModel = require("../model/userShema");
+const jwt = require("jsonwebtoken");
+const bcrypt = require("bcrypt");
+
+const signup = async (req, res) => {
+  try {
+    const { username, email, password } = req.body;
+    // input validation field
+    if (!username || !email || !password) {
+        return res.status(400).json({ error: 'All fields are required' });
+    }
+    const user = await userModel.findOne({ email: email });
+    if (user) res.status(200).json({ message: "user already present" });
+    else {
+      const hashedPassword = await bcrypt.hash(password, 10);
+
+      const user = new userModel({
+        username,
+        email,
+        password: hashedPassword,
+      });
+      await user.save();
+      res.status(200).json({ message: "user registered successfully" });
```

Comment:
Send token here so that users don't need to login to generate token, here only we can send it.